netshare/learn/setup_per_chunk_data.py

```diff
@@ -0,0 +1,385 @@
+            this_df.columns = [f"{field.column}_{i}" for i in range(this_df.shape[1])]
+            new_field_list += list(this_df.columns)
+            new_df = pd.concat([new_df, this_df], axis=1)
+
+        # Categorical field: (string | integer)
+        if "categorical" in field.get("encoding", ""):
+            this_df = pd.DataFrame(
+                field_instance.normalize(original_df[field.column].to_numpy())
+            )
+            this_df.columns = [f"{field.column}_{i}" for i in range(this_df.shape[1])]
+            new_field_list += list(this_df.columns)
+            new_df = pd.concat([new_df, this_df], axis=1)
+
+        # Continuous Field: (float)
+        if field.type == "float":
+            for column in field.get("columns") or [field.column]:
+                new_df[column] = field_instance.normalize(
+                    original_df[column].to_numpy().reshape(-1, 1)
+                )
+                new_field_list.append(column)
+
+        if "regex" in field:
+            for column in field.get("columns") or [field.column]:
+                normalized_df = pd.DataFrame(
+                    field_instance.normalize(
+                        original_df[column].to_numpy().reshape(-1, 1)
+                    )
+                )
+                normalized_df.columns = [
+                    f"{field_instance.name}_{i}" for i in range(normalized_df.shape[1])
+                ]
+                new_field_list += list(normalized_df.columns)
+                new_df = pd.concat([new_df, normalized_df], axis=1)
+
+    return new_df, new_field_list
+
+
+def write_chunk_data(
+    df_per_chunk: pd.DataFrame,
+    cross_chunks_data: CrossChunksData,
+    data_attribute: np.array,
+    data_feature: np.array,
+    chunk_id: int,
+) -> None:
+    """
+    This function writes the data of a single chunk using the learn_api.
+
+    TODO: Why do we store the cross_chunks_data for every chunk?
+        It creates trouble later when we try to load it in the postprocess phase.
+    """
+    learn_api.create_dirs(chunk_id)
+    learn_api.write_raw_chunk(df_per_chunk, chunk_id)
+    learn_api.write_data_train_npz(
+        data_attribute=data_attribute,
+        data_feature=data_feature,
+        global_max_flow_len=cross_chunks_data.global_max_flow_len,
+        chunk_id=chunk_id,
+    )
+    learn_api.write_attributes(cross_chunks_data.session_key_fields, chunk_id)
+    learn_api.write_features(cross_chunks_data.timeseries_fields, chunk_id)
+
+
+def apply_timestamp_generation(
+    df_per_chunk: pd.DataFrame,
+    cross_chunks_data: CrossChunksData,
+    new_timeseries_list: List[str],
+) -> np.array:
+    """
+    This function reads the configuration from pre_post_processor.config.timestamp,
+        and generate the normalized timestamp column if needed.
+
+    There are two types of timestamp generation:
+    1. encoding = raw: Taking a specific column as another timestamp column. TODO: I don't understand why this is needed.
```

Comment:
It's just one method we have tried in the past :)

Reply:
I see,
Then I'll delete it if this is ok with you.
I prefer to no leave unused code in the codebase :)

Reply:
I may choose to keep it as it gives another option to the user to encode the timestamp.